pennylane_qiskit/qiskit_device2.py

```diff
@@ -195,7 +195,12 @@ def reorder_fn(res):
 
         result = dict(zip(flattened_indices, flattened_results))
 
-        return tuple(result[i] for i in sorted(result.keys()))
+        result = tuple(result[i] for i in sorted(result.keys()))
+
+        # Need to check for shot_vector attribute because if an array of shots is inputted
+        # an array of results is expected by the device. Unfortunately, you cannot just
+        # "tell" the device we are not going to run an array of shots, so this is a bit of a hack
+        return result[0] if len(result) == 1 and len(tape.shots.shot_vector) == 1 else result
```

Comment:
Right now, this solution has a slightly buggy edge case:
If the User initializes a circuit with `use_primitives=True`, and that circuit only has 1 measurement AND that user set an array of `shots`, then the returned result will have the wrong format (it'll have an extra bracket around it). I cannot return result[0] because when the circuit is initialized with an array of shots, it expects an iterable. (You can see this error if you delete `and len(tape.shots.shot_vector) == 1` and run the test cases. Basically, since we are already ignoring the user when they set an array of shots (we just use the total number of shots), I think we could raise the warning here instead and also explain that their result MIGHT be wrong in the case that they set only one measurement in their circuit. Alternatively, we just raise an error and tell them this is a bad idea anyway because Qiskit doesn't do an array of shots. I'm open to either solution, or if someone has a better solution where I can avoid all of this that would be good too.

Reply:
There's a lot going on at once in this device and we aren't really able to support `shot_vector` support anyway, so I'm in favour of raising an error if we encounter one. If it's something we want maybe it can be added in later but I'd say it's pretty low priority.

Reply:
I'll raise an error for now.